test/unittest/unittest_nntrainer_tensor.cpp

```diff
@@ -1637,6 +1637,396 @@ TEST(nntrainer_Tensor, sum_02_p) {
+              status = ML_ERROR_RESULT_OUT_OF_RANGE;
+              goto end_test;
+            }
+          }
+        }
+      }
+    }
+  }
+
+  // Test for alpha == 2 and beta == 2
+  {
+    float ans_0_2_2[1][channel][height][width] = {
+      {{{128, 136, 144, 152, 160, 168, 176, 184, 192, 200}},
+       {{208, 216, 224, 232, 240, 248, 256, 264, 272, 280}}}};
+
+    float ans_1_2_2[batch][1][height][width] = {
+      {{{26, 32, 38, 44, 50, 56, 62, 68, 74, 80}}},
+      {{{126, 132, 138, 144, 150, 156, 162, 168, 174, 180}}},
+      {{{226, 232, 238, 244, 250, 256, 262, 268, 274, 280}}}};
+
+    float ans_2_2_2[batch][channel][1][width] = {
+      {{{4, 8, 12, 16, 20, 24, 28, 32, 36, 40}},
+       {{44, 48, 52, 56, 60, 64, 68, 72, 76, 80}}},
+      {{{84, 88, 92, 96, 100, 104, 108, 112, 116, 120}},
+       {{124, 128, 132, 136, 140, 144, 148, 152, 156, 160}}},
+      {{{164, 168, 172, 176, 180, 184, 188, 192, 196, 200}},
+       {{204, 208, 212, 216, 220, 224, 228, 232, 236, 240}}}};
+
+    float ans_3_2_2[batch][channel][height][1] = {
+      {{{112}}, {{314}}}, {{{516}}, {{718}}}, {{{920}}, {{1122}}}};
+
+    nntrainer::Tensor output_0_2_2(1, channel, height, width);
+    {
+      const int batch = 1;
+      GEN_TEST_INPUT(output_0_2_2, i * (channel * height * width) +
+                                     j * (height * width) + k * (width) + l +
+                                     1);
+    }
+    nntrainer::Tensor output_1_2_2(batch, 1, height, width);
+    {
+      const int channel = 1;
+      GEN_TEST_INPUT(output_1_2_2, i * (channel * height * width) +
+                                     j * (height * width) + k * (width) + l +
+                                     1);
+    }
+    nntrainer::Tensor output_2_2_2(batch, channel, 1, width);
+    {
+      const int height = 1;
+      GEN_TEST_INPUT(output_2_2_2, i * (channel * height * width) +
+                                     j * (height * width) + k * (width) + l +
+                                     1);
+    }
+    nntrainer::Tensor output_3_2_2(batch, channel, height, 1);
+    {
+      const int width = 1;
+      GEN_TEST_INPUT(output_3_2_2, i * (channel * height * width) +
+                                     j * (height * width) + k * (width) + l +
+                                     1);
+    }
+    nntrainer::Tensor result_0_2_2 = input.sum(0, output_0_2_2, 2, 2);
+    nntrainer::Tensor result_1_2_2 = input.sum(1, output_1_2_2, 2, 2);
+    nntrainer::Tensor result_2_2_2 = input.sum(2, output_2_2_2, 2, 2);
+    nntrainer::Tensor result_3_2_2 = input.sum(3, output_3_2_2, 2, 2);
+
+    for (unsigned int i = 0; i < result_0_2_2.batch(); ++i) {
+      for (unsigned int l = 0; l < result_0_2_2.channel(); ++l) {
+        for (unsigned int j = 0; j < result_0_2_2.height(); ++j) {
+          for (unsigned int k = 0; k < result_0_2_2.width(); ++k) {
+            if (ans_0_2_2[i][l][j][k] != result_0_2_2.getValue(i, l, j, k)) {
```

Comment:
this is float so we can't compare like this. 

Tensor has operator== defined so we can use this directly.

Reply:
Done